client/src/components/relatedItemsAndComparison/ProductCard.jsx

```diff
@@ -1,77 +1,110 @@
+  const [mainImg, setMainImg] = useState('');
 
-  //get info for the current related product (needed for )
-  const getRelatedProduct = () => {
-    if (productId) { //had to add the condition for initial render when no
-                     //productId was provided; adjust when change useEffect
-                     //dependencies
-      axios.get(`/products/${productId}`)
-        .then((product) => {
-          setRelatedProduct(product.data);
+  const getRating = () => {
+    if (productId) {
+      axios.get(`/meta/${productId}`)
+        .then((meta) => {
+          console.log(meta);
+          // setRating(averageRating(meta));
         })
-        .catch((err) => console.error(err))
-        .then(() => getDefaultStyles());
+        .catch((err) => console.error(err));
     }
-  }
+  };
 
-  //get default style for the current related product (needed for images and price)
-  const getDefaultStyles = () => {
-    if (productId) {  //had to add the condition for initial render when no
-                      //productId was provided; adjust when change useEffect
-                      //dependencies
-      axios.get(`/products/${productId}/styles`)
+  // get default style for the current related product (needed for images and price)
+  const getStyles = () => {
+    axios.get(`/products/${productId}/styles`)
       .then((product) => {
-        let defaultStyles = product.data.results.filter((defaultStyle) => defaultStyle['default?']);
-        if (defaultStyles.length > 0) {
-          setDefaultStyle(defaultStyles[0]);
+        const defaultStyles = product.data.results.filter((style) => style['default?']);
+        const newDefaultStyle = defaultStyles.length > 0
+          ? defaultStyles[0] : product.data.results[0];
+        setDefaultStyle(newDefaultStyle);
+
+        if (newDefaultStyle.photos[0].thumbnail_url !== null) {
+          setImages(newDefaultStyle.photos);
         } else {
-          setDefaultStyle(product.data.results[0]);
+          setImages([{
+            thumbnail_url: 'https://www.freeiconspng.com/uploads/no-image-icon-6.png',
+            url: 'https://www.freeiconspng.com/uploads/no-image-icon-6.png',
+          }]);
         }
       })
       .catch((err) => console.error(err));
-    }
-  }
+  };
+
+  // get info for the current related product (needed for )
+  const getRelatedProduct = () => {
+    axios.get(`/products/${productId}`)
+      .then((product) => {
+        setRelatedProduct(product.data);
+        // getRating();
+      })
+      .catch((err) => console.error(err))
+      .then(() => getStyles());
+  };
 
+  useEffect(() => {
+    if (productId) {
+      getRelatedProduct();
+    }
+    // getStyles();
+  }, [productId]);
 
   useEffect(() => {
-    getRelatedProduct();
-    // getDefaultRelatedProduct();
-  }, []);
+    if (images.length > 0) {
+      setMainImg(images[0].thumbnail_url);
+    }
+  }, [images]);
+
+  // const handleNavigate = () => {
```

Comment:
You should go through this file and get rid of unnecessary commented code

Reply:
Fair, I'll make sure to remove them before the next PR. I'm not planning on keeping the comments, just left them to remind myself of the missing functionality.